compiler/frontend/src/mir/body.rs

```diff
@@ -411,7 +411,53 @@ impl BodyBuilder {
             responsible,
         )
     }
+    pub fn push_if_not<E>(
+        &mut self,
+        hir_id: &hir::Id,
+        condition: Id,
+        else_builder: E,
```

Comment:
You can inline the constraint so there are no type parameters anymore.
```suggestion
        else_builder: impl FnOnce(&mut Self),
```

Reply:
Should I also apply this to the rest of this class (I was matching the style of other methods in this class)?

Reply:
Maybe later?

Reply:
I think we should eventually use it where applicable, but that can be done over time. I think the feature is relatively recent, so we only used it for new code and didn't update everything yet